pkgs/applications/video/handbrake/default.nix

```diff
@@ -30,35 +30,62 @@
   # for now we disable GTK GUI support on Darwin. (It may be possible to remove
   # this restriction later.)
   useGtk ? !stdenv.isDarwin, wrapGAppsHook ? null,
-                             intltool ? null,
-                             glib ? null,
-                             gtk3 ? null,
-                             libappindicator-gtk3 ? null,
-                             libnotify ? null,
-                             gst_all_1 ? null,
-                             dbus-glib ? null,
-                             udev ? null,
-                             libgudev ? null,
-                             hicolor-icon-theme ? null,
+  intltool ? null,
+  glib ? null,
+  gtk3 ? null,
+  libappindicator-gtk3 ? null,
+  libnotify ? null,
+  gst_all_1 ? null,
+  dbus-glib ? null,
+  udev ? null,
+  libgudev ? null,
+  hicolor-icon-theme ? null,
   # FDK
   useFdk ? false, fdk_aac ? null
 }:
 
-assert stdenv.isDarwin -> AudioToolbox != null && Foundation != null
-  && libobjc != null && VideoToolbox != null;
+assert stdenv.isDarwin -> AudioToolbox != null
+       && Foundation != null
+       && libobjc != null
+       && VideoToolbox != null;
 
 stdenv.mkDerivation rec {
   pname = "handbrake";
   version = "1.3.2";
 
-  src = fetchurl {
-    #  2020-05-05: NOTE: Thou fetching from GitHub, still fetchurl required,
-    #  because this tarball has their "special" packaging and so
-    #  internal "special" version information
-    url = ''https://github.com/HandBrake/HandBrake/releases/download/${version}/HandBrake-${version}-source.tar.bz2'';
-    sha256 = "0w7jxjrccvxp7g15dv0spildg5apmqp4gwbcqmg58va2gylynvzc";
+  src = fetchFromGitHub {
+    owner = "HandBrake";
+    repo = "HandBrake";
+    rev = version;
+    sha256 = "04z3hcy7m5yvma849rlrsx2wdqmkilkl1qds9yrzr2ydpw697f85";
+    extraPostFetch = ''
+      echo "DATE=$(date +"%F %T %z" -r $out/NEWS.markdown)" > $out/version.txt
```

Comment:
Wait are we sure this is reproducible? It will be only if the "modtime" of the file stays the same? Maybe try:

https://github.com/HandBrake/HandBrake/blob/0b6b6847631881bd16027c90b079c06adadf666f/make/configure.py#L890

Reply:
> Wait are we sure this is reproducible?

Yep.

> It will be only if the "modtime" of the file stays the same?

And it will at least until a new version is installed in which case we want it to change.

Running `repo-info.sh` doesn't work as we don't have a full git checkout - this is the whole reason for generating version.txt manually.

Reply:
> Running repo-info.sh doesn't work as we don't have a full git checkout - this is the whole reason for generating version.txt manually.

You will be able to run that script if you'll tell `fetchFromGitHub` to `leaveDotgit = true` (I think this is how it's called) and you'll also be able to get rid of the other `echo` commands then. I think this is much cleaner so it's worth a try.